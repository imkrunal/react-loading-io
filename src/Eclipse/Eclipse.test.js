import React from 'react'
import serializer from 'jest-emotion'
import renderer from 'react-test-renderer'
import { Eclipse } from '..'

expect.addSnapshotSerializer(serializer)

it('eclipse renders correctly', () => {
  const tree = renderer.create(<Eclipse />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('px converts', () => {
  const tree = renderer.create(<Eclipse size={10} sizeUnit="em" />).toJSON()
  expect(tree).toMatchSnapshot()
})
