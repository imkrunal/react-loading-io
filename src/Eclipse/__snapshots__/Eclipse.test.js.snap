// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eclipse renders correctly 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-1 {
  position: relative;
  width: 200px !important;
  height: 200px !important;
  -webkit-transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
  -ms-transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
  transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
}

.emotion-0 {
  position: absolute;
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  width: 160px;
  height: 160px;
  top: 20px;
  left: 20px;
  border-radius: 50%;
  box-shadow: 0 4px 0 0 #f08d43;
  -webkit-transform-origin: 80px 82px;
  -ms-transform-origin: 80px 82px;
  transform-origin: 80px 82px;
}

<div>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    />
  </div>
</div>
`;

exports[`px converts 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  position: absolute;
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  width: 160px;
  height: 160px;
  top: 20px;
  left: 20px;
  border-radius: 50%;
  box-shadow: 0 4px 0 0 #f08d43;
  -webkit-transform-origin: 80px 82px;
  -ms-transform-origin: 80px 82px;
  transform-origin: 80px 82px;
}

.emotion-1 {
  position: relative;
  width: 10em !important;
  height: 10em !important;
  -webkit-transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
  -ms-transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
  transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
}

<div>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    />
  </div>
</div>
`;
